Extra character left when deleting words in `sh` files

### Summary

This seems to only happen in `sh` files. Whenever I hit `Ctrl-w` to delete words, the last character until the cursor position is always left untouched, making it impossible to, say, delete everything before the cursor by repeating `Ctrl-w`.

### Reproduction Steps

**I tried this:**

1. Creating an empty `sh` file and opening it with no local config:

```console
$ touch test.sh
$ hx -c /dev/null test.sh
```

2. Hit the following keys: `i` `a` `b` `c`. We're now in insert mode with the word `abd`.

3. Hit `Ctrl-w` to delete the word.

**I expected this to happen:**

We should have an empty buffer now, since the word should be deleted.

**Instead, this happened:**

The text in the buffer becomes `c`.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2022-10-17T10:57:55.139 helix_view::editor [ERROR] Failed to initialize the LSP for `source.bash` { cannot find binary path }
```

</details>


### Platform

Linux

### Terminal Emulator

tmux 3.3a

### Helix Version

22.08.1-229-g09a6df19

--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -2880,7 +2880,7 @@ pub mod insert {
 
     /// Exclude the cursor in range.
     fn exclude_cursor(text: RopeSlice, range: Range, cursor: Range) -> Range {
-        if range.to() == cursor.to() {
+        if range.to() == cursor.to() && text.len_chars() != cursor.to() {
             Range::new(
                 range.from(),
                 graphemes::prev_grapheme_boundary(text, cursor.to()),

